examples/template_alignment.py

```diff
@@ -0,0 +1,179 @@
+
+#############################################################################
+# Compute the average of training subjects images, to be used as a baseline.
+# -------------------------------------------------------------------------
+# We create an image with as many contrasts as any subject
+# representing for each contrast the average of all train subjects maps.
+import numpy as np
+
+masked_imgs = [masker.transform(img) for img in template_train]
+average_img = np.mean(masked_imgs, axis=0)
+average_subject = masker.inverse_transform(average_img)
+
+#############################################################################
+# Create a template from the training subjects.
+# ----------------------------------------------
+# We define an estimator using the class TemplateAlignment:
+# * We align the whole brain through 'multiple' local alignments.
+# * These alignments are calculated on a parcellation of the brain in 150\
+#   pieces, this parcellation creates group of functionnally similar voxels.
+# * The template is created iteratively, aligning all subjects data into a
+#   common space, from which the template is inferred and aligning again to this
+#   new template space.
+
+from fmralign.template_alignment import TemplateAlignment
+from nilearn.image import index_img
+
+template_estim = TemplateAlignment(
+    n_pieces=150, alignment_method='ridge_cv', mask=masker)
+template_estim.fit(template_train)
+
+#############################################################################
+# Predict subject sub-07 PA data from the template and fitted estimator
+# ----------------------------------------------------------------
+# We use target_train data to fit the transform, indicating it corresponds to
+# the contrasts indexed by train_index and predict from this learnt alignment
+# contrasts corresponding to template test_index numbers.
+
+# For each train subject and for the template, the AP contrasts are sorted from
+# 0, to 53, and then the PA contrasts from 53 to 106.
+train_index = range(53)
+test_index = range(53, 106)
+# We input the mapping image target_train in a list, we could have input more
+# than one subject for which we'd want to predict : [train_1, train_2 ...]
+prediction_from_template = template_estim.transform([target_train], train_index,
+                                                    test_index)
+# As a baseline prediction, let's just take the average of activations across subjects.
+prediction_from_average = index_img(average_subject, test_index)
+
+#############################################################################
+# Defining a metric to score the performance of these predictions
+# ---------------------------------------------------
+# We define a scoring function to measure the correlation, between
+# the prediction and the ground truth for each voxel profiles of activation.
+
+import numpy as np
+from scipy.stats import pearsonr
+
+
```

Comment:
I believe that such a metric cold be used elsewhere. Hence I suggest to integrate it to the library, e.g. in an `_utils.py` module.

Reply:
And the, add a test ;-)